mne/inverse_sparse/mxne_optim.py

```diff
@@ -714,26 +714,141 @@ def norm_l1_tf(Z, shape, n_orient):
     return l1_norm
 
 
-@verbose
-def _tf_mixed_norm_solver_bcd_(M, G, Z, active_set, alpha_space, alpha_time,
-                               lipschitz_constant, phi, phiT,
-                               wsize=64, tstep=4, n_orient=1,
-                               maxit=200, tol=1e-8, log_objective=True,
-                               perc=None, verbose=None):
+def prox_l1_full(Y, alpha, n_orient):
+    """proximity operator for l1 norm with multiple orientation support
+
+    L2 over orientation and L1 over position (space + time)
+
+    Example
+    -------
+    >>> Y = np.tile(np.array([1, 2, 3, 2, 0], dtype=np.float), (2, 1))
+    >>> Y = np.r_[Y, np.zeros_like(Y)]
+    >>> print(Y)
+    [[ 1.  2.  3.  2.  0.]
+     [ 1.  2.  3.  2.  0.]
+     [ 0.  0.  0.  0.  0.]
+     [ 0.  0.  0.  0.  0.]]
+    >>> Yp, active_set = prox_l1(Y, 2, 2)
+    >>> print(Yp)
+    [[ 0.          0.58578644  1.58578644  0.58578644  0.        ]
+     [ 0.          0.58578644  1.58578644  0.58578644  0.        ]]
+    >>> print(active_set)
+    [ True  True False False]
+    """
+    n_positions = Y.shape[0] // n_orient
+    norms = np.sqrt((Y * Y.conj()).real.T.reshape(-1, n_orient).sum(axis=1))
+    shrink = np.maximum(1.0 - alpha / np.maximum(norms, alpha), 0.0)
+    shrink = shrink.reshape(-1, n_positions).T
+    active_set = np.any(shrink > 0.0, axis=1)
+    if n_orient > 1:
+        active_set = np.tile(active_set[:, None], [1, n_orient]).ravel()
+    for o in range(n_orient):
+        Y[o::n_orient] *= shrink
+    return Y
+
+
+def dgap_l21l1(M, G, Z, active_set, alpha_space, alpha_time, phi, phiT, shape,
+               n_orient, highest_d_obj, screening):
+    """Duality gaps for the time-frequency mixed norm inverse problem
+
+    For details see:
+    Wang J., and Ye J.,
+    Two-layer feature reduction for sparse-group lasso via decomposition of
+    convex sets, in Advances in Neural Information Processing Systems (NIPS),
+    vol. 27, pp. 2132-2140, 2014.
+
+    Bach F., Jenatton R., Mairal J., and Obozinski G.,
+    Optimization with sparsity-inducing penalties, Foundations and Trends
+    in Machine Learning, vol. 4, no. 1, pp. 1-106, 2012.
+
+    Gramfort A., Kowalski M., and Hamalainen, M.,
+    Mixed-norm estimates for the M/EEG inverse problem using accelerated
+    gradient methods, Physics in Medicine and Biology, 2012
+    http://dx.doi.org/10.1088/0031-9155/57/7/1937
+
+    Parameters
+    ----------
+    M : array of shape [n_sensors, n_times]
+        The data.
+    G : array of shape [n_sensors, n_active]
+        Gain matrix a.k.a. lead field.
+    Z : array of shape [n_active, n_coefs]
+        Sources in TF domain.
+    active_set : array of bool
+        Mask of active sources.
+    alpha_space : float
+        The spatial regularization parameter. It should be between 0 and 100.
```

Comment:
in this function nothing requires that alpha is between 0 and 100 so I'd remove the second sentence